### PR TITLE
Run Copr builds directly and use LLMs only for diagnosis

### PR DESCRIPTION
**Pull request by opohorel**
Call the existing build tool deterministically so successful builds no longer require model orchestration. Share the structured Copr result schema and invoke a diagnosis-only agent for failed builds with logs.

Migrate backport, inheritance, rebase, MR updates, and consolidation while preserving their timeout and retry policies. Keep rebuild and the incremental backport repair loop unchanged.

Add regression tests for gateway decoding, zero-LLM success handling, error classification, cancellation, and diagnosis without resubmission. Document the shared build flow and its dry-run behavior.

Validation: 64 focused container tests passed; Ruff lint and format checks passed. Broader container testing found two gateway startup test failures and the same missing-MCP-settings error in the install smoke test. The gateway test failures also reproduce on unchanged commit 00d2ba6. Other component suites passed.

Assisted-by: Codex

I've ran e2e test for build succeeding on first try and build failing on the first time with the need to fix it. No regressions were found. GPT-6 Astra estimates around 8% of token cost savings per backport task.

### CODE REVIEW

File: ymir/agents/build_agent.py [hunk 1]
```diff
@@ -1,32 +1,108 @@
...
+        logger.warning("Copr build tool failed for %s: %s", build_input.jira_issue, e)
+        return BuildOutputSchema(success=False, error=str(e), is_infra_error=True)
+
+    # MCP normally returns a dictionary; also support text-only JSON results.
+    # Malformed responses must raise rather than being interpreted as success.
+    result = (
+        BuildResult.model_validate_json(raw_result)
+        if isinstance(raw_result, str)
+        else BuildResult.model_validate(raw_result)
+    )
```
Comment (nforro) [inline ymir/agents/build_agent.py]: Shouldn't this be rather handled in `run_tool()`?

File: ymir/agents/build_agent.py [hunk 2]
```diff
@@ -1,32 +1,108 @@
...
+
+logger = logging.getLogger(__name__)
+
+
+async def run_build(
+    *,
+    build_input: BuildInputSchema,
+    available_tools: list[Tool],
+    local_tool_options: dict[str, Any],
+) -> BuildOutputSchema:
```
Comment (nforro) [inline ymir/agents/build_agent.py:53]: I realize it's an overkill, but wouldn't it be better to use a `Workflow` here? For consistency and future-proofing.

**Review by nforro — COMMENTED** [review]
Just curious, what was the motivation for this change?

**Comment by opohorel** [comment]
> Just curious, what was the motivation for this change?

mainly to reduce token spend on a workflow, where LLM isn't that needed. thanks for the comments, I'll make the changes.

**Review by nforro — COMMENTED** [review]
LGTM, a couple of LLM-identified issues:


1. **High** `ymir/agents/mr_consolidation_agent.py:786`: The fallback Copr project name uses the package verbatim. Valid RPM names such as `compat-libstdc++-33` contain `+`, which Copr rejects. Sanitize and length-limit the generated identifier.

2. **Medium** `ymir/tools/privileged/copr.py:26`: `ymir-tools` now imports the newly added `BuildResult`, but still declares `ymir-common>=0.1.0`. Upgrading tools with an older common package causes an import failure. Bump the common version and minimum dependency.

3. **Medium** `ymir/agents/build_agent.py:110`: `except Exception` catches BeeAI `AbortError`, which inherits both `FrameworkError` and `asyncio.CancelledError`. Framework-originated cancellation can therefore be treated as an ordinary diagnosis failure. Add `except asyncio.CancelledError: raise` first and test `AbortError`.

**Review by nforro — APPROVED** [review]
